logprep/ng/event/event_state.py

```diff
@@ -94,6 +94,41 @@ def __init__(self) -> None:
 
         self.current_state: EventStateType = cast(EventStateType, EventStateType.RECEIVING)
 
+    def __eq__(self, other: object) -> bool:
+        """
+        Check equality with another EventState instance.
+
+        Parameters
+        ----------
+        other : object
+            The object to compare against.
+
+        Returns
+        -------
+        bool
+            True if the other object is an EventState with the same current state
+            or an EventStateType matching the current state, otherwise False.
+        """
+
+        if not isinstance(other, (EventState, EventStateType)):
+            return False
+        try:
+            return self.current_state == other.current_state
+        except AttributeError:
+            return self.current_state == other
```

Comment:
Please use this instead:
```suggestion
    if isinstance(other, EventState):
        return self.current_state == other.current_state
    return self._current_state == other
```
**Note**: _current_state has to be renamed